### DISCUSSION TITLE
模块联邦依赖共享到底有没有生效？

### DISCUSSION THREAD

**Discussion by zhtyytg** [web-infra-dev/rsbuild#1945] [General]
https://rsbuild.dev/zh/config/module-federation/options 官方文档最下面贴了两个例子。
![image](https://github.com/web-infra-dev/rsbuild/assets/42018740/bdc8b6aa-db57-42ff-addb-4ea4f1f830e2)

分别拉下来不修改配置内容，跑通后控制台看react加载结果类似下面这种（两个项目不一样但表现形式相近）
![1711521576591](https://github.com/web-infra-dev/rsbuild/assets/42018740/3ae97b26-bb18-4845-aaf0-2070c7853b87)
即使配置了react shared，依然还是有至少部分react依赖被加载了。可以认为这部分react依赖是必须要被每个remote自身单独加载吗？是否会对singleton单实例造成影响？

**Comment by chenjiahan** [comment]
The issue of lib-react.js being repeatedly loaded in the screenshot has been resolved, and a fix version will be released today.

Reference:

> 截图中 lib-react.js 重复加载的问题已经修复了，今天会发布一个修复版本。
> 背景参考：

- https://github.com/web-infra-dev/rsbuild/pull/1942
- https://github.com/module-federation/module-federation-examples/issues/3161

**Reply by zhtyytg** [reply]
已经看到了修复：拆分了react依赖
![image](https://github.com/web-infra-dev/rsbuild/assets/42018740/57e333fb-1f0f-499e-80fe-10d4794b5242)

打包后可以看到host和remote都具有依赖如下：
886: react-jsx-runtime.production.min.js
170: react-dom.production.min.js && scheduler.production.min.js
667: react.production.min.js

实际上host只加载了react-jsx-runtime.production.min.js，而react.production.min.js是从remote加载过来的
![image](https://github.com/web-infra-dev/rsbuild/assets/42018740/fa773aca-6b72-4fde-adb3-1f301ed58c9c)

为什么会出现同属react的依赖却分别从host和remote加载的情况呢？
